torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -400,6 +401,7 @@ def __init__(
         self,
         module: nn.Module,
         process_group: ProcessGroupType = None,
+        device_mesh: DeviceMesh = None,
```

Comment:
This typing is not correct, same as `process_group`. There should be `Optional[]`.